Sources/Core/AccessTokenWallet.swift

```diff
@@ -27,6 +27,13 @@ class AccessTokenWallet {
   let graphConnectionProvider: GraphConnectionProviding
   let graphRequestPiggybackManager: GraphRequestPiggybackManaging.Type
 
+  /**
+   The global shared instance of AccessTokenWallet
+
+   Overriding this value may result in unexpected and unwanted behavior
+  */
+  static var shared: AccessTokenWallet = AccessTokenWallet()
```

Comment:
Or

```suggestion
  static let shared: AccessTokenWallet = AccessTokenWallet()
```

Reply:
I am sort of against this but expecting to and willing to lose the argument on it. 

Here are my arguments for keeping it internal:
1. We're working with kits so as long as we don't declare it public we shouldn't have to worry about an external dev setting it.
2. It defers some decisions about dependency management. Once we start testing more complex scenarios we should still be able to inject a wallet where needed (in the initializer say) but this gets onerous.
3. It makes it easier to manage the object's lifecycle and write code that can reset the shared wallet instance (may on logout etc...)

Would an internal designated setter be a better solution?

Reply:
Changed it to let for now since it does not need setting or cleanup but would like to revisit this in general terms going forward since it will no doubt come up in some other form. 👍